discussion.md

```diff
@@ -3,4 +3,131 @@ layout: page
+That file is where the path to a remote location that the repo may have been cloned from is stored.
+You can open that file for editing with the command `hg config --local`.
+
+
+## Non-text Files
+
+Recall when we discussed [Conflicts](03-conflict.html)
+there was a challenge that asked:
+
+> What does hg do
+> when there is a conflict in an image or some other non-textual file
+> that is stored in version control?
+
+We will now revisit this in more detail.
+
+Many people want to version control non-text files, such as images, PDFs and Microsoft Office or LibreOffice documents.
+It is true that Mercurial can handle these filetypes (which fall under the banner of "binary" file types).
+However, just because it *can* be done doesn't mean it *should* be done.
+
+Much of Mercurial's magic comes from being able to do line-by-line comparisons ("diffs") between files.
+This is generally easy for programming source code and marked up text.
+For non-text files, a diff can usually only detect that the files have changed
+but can't say how or where.
+
+This has various impacts on Mercurial's performance and will make it difficult to
+compare different versions of your project.
+
+For a basic example to show the difference it makes,
+we're going to go see what would have happened if Dracula had tried
+using outputs from a word processor instead of plain text.
+
+Create a new directory and go into it:
+
+~~~ {.bash}
+$ mkdir planets-nontext
+$ cd planets-nontext
+~~~
+
+Use a program such as Microsoft Word or LibreOffice Writer to create a new document.
+Enter the same text that we began with before:
+
+~~~ {.output}
+Cold and dry, but everything is my favorite color
+~~~
+
+Save the document into the `planets-nontext` directory with the name of `mars.doc`.
+Back in the terminal, run the usual commands for setting up a new Mercurial repository:
+
+~~~ {.bash}
+$ hg init
+$ hg add mars.doc
+$ hg commit -m "Starting to think about Mars"
+~~~
+
+Then make the same changes to `mars.doc` that we (or Vlad) previously made to `mars.txt`.
+
+~~~ {.output}
+Cold and dry, but everything is my favorite color
+The two moons may be a problem for Wolfman
+~~~
+
+Save and close the word processor.
+Now see what Git thinks of your changes:
+
+~~~ {.bash}
+$ hg diff
+~~~
+~~~ {.output}
+diff -r 0f3937fd3863 mars.odt
+Binary file mars.odt has changed
+~~~
+
+Compare this to the earlier `git diff` obtained when using text files:
+
+~~~ {.output}
+diff -r 72ab25fa99a1 mars.txt
+--- a/mars.txt  Mon Apr 14 14:41:58 2014 -0400
++++ b/mars.txt  Mon Apr 14 15:48:53 2014 -0400
+@@ -1,1 +1,2 @@
+ Cold and dry, but everything is my favorite color
++The two moons may be a problem for Wolfman
+~~~
+
+Notice how plain text files give a much more informative diff.
+You can see exactly which lines changed and what the changes were.
+
+An uninformative `hg diff` is not the only consequence of using Mercurial on binary files.
```

Comment:
Here you could mention that a new copy has to be stored whenever a binary file changes and hence much more space is needed.